packages/amplify-graphql-transformer-core/src/utils/provision-strategy-utils.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  DatasourceProvisionConfig,
+  DatasourceProvisionStrategy,
+  TransformerBeforeStepContextProvider,
+} from '@aws-amplify/graphql-transformer-interfaces';
+
+/**
+ * Get the project-level datasource provision strategy. When typeName is provided, the model level strategy will be fetched
+ * @param ctx transfomer before step context
+ * @param typeName model name defined in GraphQL schema defintion
+ * @returns Datasource provision strategy for the provided model. Undefined if not found
+ */
+export function getDatasourceProvisionStrategy(
```

Comment:
This shouldn't return `undefined`; instead we should ensure that we assign the default dynamoDB model strategy to each model, and then we can be sure that each model will use the default DynamoDB provisioning strategy unless otherwise overridden.